tests/gds_kernel_latency.c

```diff
@@ -234,15 +237,15 @@ static struct pingpong_context *pp_init_ctx(struct ibv_device *ib_dev, int size,
                         ctx->txbufexp_addr = (uintptr_t*)gpu_malloc(page_size, sizeof(uintptr_t)*max_batch_len);
                 }
         } else {
-                ctx->txbuf = memalign(page_size, ctx->txtot_size); //posix_memalign
-                ctx->rxbuf = memalign(page_size, ctx->rxtot_size);
+                assert(0 == posix_memalign((void **)&(ctx->txbuf), page_size, ctx->txtot_size));
```

Comment:
why switching from memalign to posix_memalign ?

Reply:
because memalign is obsolete https://linux.die.net/man/3/memalign

Reply:
good point

Reply:
FYI judging from https://github.com/linux-rdma/rdma-core/blob/1cf909a14b3d07c8a301e3de03bfb91e62aaeff5/libibverbs/examples/ud_pingpong.c#L309 it looks like memalign is still being used.
Here we forked that code, so it is up to us.
Note that switching to posix_memalign() brings a bit different requirements wrt memalign():
"The function posix_memalign() allocates size bytes and places the address of the allocated memory in *memptr. The address of the allocated memory will be a multiple of alignment, which must be a power of two and a multiple of sizeof(void *)."
Also note that buffer size is not checked to be a multiple of alignment.